File: HW_5/tail_recursive.py
def findMinValue(a, i = 0, sofar = None):
    if i == len(a) - 1:
        return a[i] if a[i] < sofar else sofar
    elif sofar == None:
        sofar = a[i]
    elif a[i] < sofar:
        sofar = a[i]
    
    return findMinValue(a, i + 1, sofar)

File: HW_5/tail_recursive.c
int minValueHelper (int * anArray, int length, int i, int sofar) {
    if (i == length - 1) {
        return  (anArray[i] < sofar) ? anArray[i] : sofar;  
    }
    else if (anArray[i] < sofar) {
        sofar = anArray[i];
    }
    
    return minValueHelper(anArray, length, i + 1, sofar);
}

int findMinValue(int * anArray, int length) {
    return minValueHelper(anArray, length, 0, anArray[0]);
}
